compiler/semexprs.nim

```diff
@@ -1094,7 +1094,7 @@ proc builtinFieldAccess(c: PContext, n: PNode, flags: TExprFlags): PNode =
     of tyObject, tyTuple:
       if ty.n != nil and ty.n.kind == nkRecList:
         for field in ty.n:
-          if field.sym.name == i:
+          if field.kind == nkSym and field.sym.name == i:
```

Comment:
Did I get it wrong? It iterates other the 'ty.n' in line 1096. Maybe it needs to iterate other all the fields of the object variants here?